### ISSUE TITLE
add `shrink_to_fit` to builders

### ISSUE THREAD

**Issue by ritchie46** [pola-rs/polars#1499]

**Comment by orlp** [comment]
I don't think this is necessary, it would be rather expensive to waste time to copy all data again, I don't see where we'd do that. The caller should do its best to provide accurate reserves.
